src/pages/iou/IOUTransactions.js

```diff
@@ -17,33 +18,105 @@ const propTypes = {
 
     /** ReportID for the associated IOU report */
     iouReportID: PropTypes.number.isRequired,
+
+    /** Email for the authenticated user */
+    userEmail: PropTypes.string.isRequired,
+
+    /** Has the iouReport been paid? */
+    isReportPaid: PropTypes.bool,
 };
 
 const defaultProps = {
     reportActions: {},
+    isReportPaid: false,
 };
 
-const IOUTransactions = ({
-    reportActions,
-    chatReportID,
-    iouReportID,
-}) => (
-    <View style={[styles.mt3]}>
-        {_.map(reportActions, (reportAction) => {
-            if (reportAction.actionName === 'IOU'
-                && reportAction.originalMessage.IOUReportID === iouReportID) {
-                return (
-                    <ReportTransaction
-                        chatReportID={chatReportID}
-                        iouReportID={iouReportID}
-                        action={reportAction}
-                        key={reportAction.sequenceNumber}
-                    />
-                );
+class IOUTransactions extends Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            rejectableTransactionIDs: [],
+        };
+
+        this.updateRejectableTransactions = this.updateRejectableTransactions.bind(this);
+    }
+
+    componentDidMount() {
+        this.updateRejectableTransactions();
+    }
+
+    componentDidUpdate(oldProps) {
+        if (oldProps.reportActions !== this.props.reportActions) {
+            this.updateRejectableTransactions();
+        }
+    }
+
+    /**
+     * Update the rejectableTransactionIDs state array. A transaction must meet multiple requirements in order to be
+     * rejectable. We must exclude transactions not associated with the iouReportID, actions which have already been
+     * rejected, and those which are not of type 'create').
+     */
+    updateRejectableTransactions() {
+        if (this.props.isReportPaid) {
+            this.setState({rejectableTransactionIDs: []});
+            return;
+        }
+
+        const actionsForIOUReport = _.filter(this.props.reportActions, action => action.originalMessage
+            && action.originalMessage.type && action.originalMessage.IOUReportID === this.props.iouReportID);
+
+        const rejectedTransactions = _.filter(actionsForIOUReport, action => ['cancel', 'decline']
+            .includes(action.originalMessage.type));
+
+        const rejectedTransactionIDs = _.map(rejectedTransactions, (rejectedTransaction) => {
+            const transactionID = lodashGet(rejectedTransaction, 'originalMessage.IOUTransactionID', 0);
+
+            // API returns transactionID as type String in certain cases (if the transaction was rejected), this
+            // workaround should be removed once this is fixed: https://github.com/Expensify/Expensify/issues/165329
+            return Number(transactionID);
+        });
```

Comment:
This won't work on Prod. 

I'm not holding this PR on that issue, due to the urgency of this PR. I am betting that I can [fix this issue ](https://github.com/Expensify/Expensify/issues/165329) before this PR is merged. IOU 10 Melvincoins if this doesn't occur.